redis/tests/support/mod.rs

```diff
@@ -98,18 +102,80 @@ impl RedisServer {
                 redis::ConnectionAddr::Unix(PathBuf::from(&path))
             }
         };
-        RedisServer::new_with_addr(addr, None, |cmd| {
-            cmd.spawn()
-                .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
-        })
+        RedisServer::new_with_addr(
+            addr,
+            None,
+            |cmd| {
+                cmd.spawn()
+                    .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
+            },
+            None,
+        )
+    }
+
+    pub fn new_with_modules(modules: Vec<RedisModules>) -> RedisServer {
+        let server_type = ServerType::get_intended();
+        let addr = match server_type {
+            ServerType::Tcp { tls } => {
+                // this is technically a race but we can't do better with
+                // the tools that redis gives us :(
+                let addr = &"127.0.0.1:0".parse::<SocketAddr>().unwrap().into();
+                let socket = Socket::new(Domain::IPV4, Type::STREAM, None).unwrap();
+                socket.set_reuse_address(true).unwrap();
+                socket.bind(addr).unwrap();
+                socket.listen(1).unwrap();
+                let listener = TcpListener::from(socket);
+                let redis_port = listener.local_addr().unwrap().port();
+                if tls {
+                    redis::ConnectionAddr::TcpTls {
+                        host: "127.0.0.1".to_string(),
+                        port: redis_port,
+                        insecure: true,
+                    }
+                } else {
+                    redis::ConnectionAddr::Tcp("127.0.0.1".to_string(), redis_port)
+                }
+            }
+            ServerType::Unix => {
+                let (a, b) = rand::random::<(u64, u64)>();
+                let path = format!("/tmp/redis-rs-test-{}-{}.sock", a, b);
+                redis::ConnectionAddr::Unix(PathBuf::from(&path))
+            }
+        };
+        RedisServer::new_with_addr(
+            addr,
+            None,
+            |cmd| {
+                cmd.spawn()
+                    .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
+            },
+            Some(modules),
+        )
     }
 
     pub fn new_with_addr<F: FnOnce(&mut process::Command) -> process::Child>(
         addr: redis::ConnectionAddr,
         tls_paths: Option<TlsFilePaths>,
         spawner: F,
+        modules: Option<Vec<RedisModules>>,
```

Comment:
I think this can just take a `&[RedisModules]` instead. If there are no modules, it can be an empty slice, so no need for the `Option` wrapper.